src/xr/xr-plane.js

```diff
@@ -0,0 +1,133 @@
+import { Vec3 } from '../math/vec3.js';
+import { Quat } from '../math/quat.js';
+
+let ids = 0;
+
+/**
+ * @class
+ * @name XrPlane
+ * @classdesc Detected Plane instance that provides position, rotation and polygon points. Plane is a subject to change during its lifetime.
+ * @description Detected Plane instance that provides position, rotation and polygon points. Plane is a subject to change during its lifetime.
+ * @param {XrPlaneDetection} planeDetection - Plane detection system.
+ * @param {object} xrPlane - XRPlane that is instantiated by WebXR system.
+ * @property {number} id Unique identifier of a plane.
+ * @property {string|null} orientation Plane's pecific orientation (horizontal or vertical) or null if orientation is anything else.
+ */
+class XrPlane extends EventHandler {
+    constructor(planeDetection, xrPlane) {
+        super();
+
+        this._id = ++ids;
+
+        this._planeDetection = planeDetection;
+        this._manager = this._planeDetection._manager;
+
+        this._xrPlane = xrPlane;
+        this._lastChangedTime = this._xrPlane.lastChangedTime;
+        this._orientation = this._xrPlane.orientation;
+
+        this._position = new Vec3();
+        this._rotation = new Quat();
+    }
+
+    /**
+     * @event
+     * @name XrPlane#remove
+     * @description Fired when {@link XrPlane} is removed.
+     * @example
+     * plane.once('remove', function () {
+     *     // plane is not available anymore
+     * });
+     */
+
+    /**
+     * @event
+     * @name XrPlane#change
+     * @description Fired when {@link XrPlane} attributes such as: orientation and/or points have been changed. Position and Rotation can change at any time without triggering `change` event.
+     * @example
+     * plane.on('change', function () {
+     *     // plane has been changed
+     * });
+     */
+
+    destroy() {
+        this.fire('remove');
+    }
+
+    update(frame) {
+        const pose = frame.getPose(this._xrPlane.planeSpace, this._manager._referenceSpace);
+        if (pose) {
+            this._position.copy(pose.transform.position);
+            this._rotation.copy(pose.transform.orientation);
+        }
+
+        // has not changed
+        if (this._lastChangedTime !== this._xrPlane.lastChangedTime) {
+            this._lastChangedTime = this._xrPlane.lastChangedTime;
+
+            // attributes have been changed
+            this.fire('change');
+        }
+    }
+
+    /**
+     * @function
+     * @name XrPlane#getPosition
+     * @description Get the world space position of a plane.
+     * @returns {Vec3} The world space position of a plane.
+     */
+    getPosition() {
+        return this._position;
+    }
+
+    /**
+     * @function
+     * @name XrPlane#getRotation
+     * @description Get the world space rotation of a plane.
+     * @returns {Vec3} The world space rotation of a plane.
+     */
+    getRotation() {
+        return this._rotation;
+    }
```

Comment:
I'm not sure about the contention in the Xr codebase, but should these be property getters instead of functions?

Reply:
These are functions to be consistent with other XR modules as well as PlayCanvas codebase, where they are functions.